### PR TITLE
Add PtrConstruct template to use in Variant constructors

### PR DESCRIPTION
**Pull request by vnen**
Since the PtrToArg::encode requires the value to be constructed previously. With PtrConstruct this is not required.

### CODE REVIEW

File: core/variant/variant_construct.cpp [hunk 1]
```diff
@@ -39,6 +39,59 @@
...
+MAKE_PTRCONSTRUCT(bool);
+MAKE_PTRCONSTRUCT(int64_t);
+MAKE_PTRCONSTRUCT(double);
+MAKE_PTRCONSTRUCT(String);
+MAKE_PTRCONSTRUCT(Vector2);
+MAKE_PTRCONSTRUCT(Rect2);
+MAKE_PTRCONSTRUCT(Vector3);
+MAKE_PTRCONSTRUCT(Vector2i);
+MAKE_PTRCONSTRUCT(Rect2i);
+MAKE_PTRCONSTRUCT(Vector3i);
```
Comment (akien-mga) [inline core/variant/variant_construct.cpp:69]: I suggest using the same order as in the `Variant::Type` enum, that makes it easier to parse the list and make sure nothing is missing.

Comment (vnen) [reply core/variant/variant_construct.cpp:69]: TBH I just copied the list from `method_ptrcall.h` but makes sense to keep the order.

Comment (akien-mga) [reply core/variant/variant_construct.cpp:69]: Might be worth changing in `method_ptrcall.h` at the same time then.

Comment (vnen) [reply core/variant/variant_construct.cpp]: Okay, I added a commit to change the other file too.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**